### DISCUSSION TITLE
如何使用docker环境调试本项目？

### DISCUSSION THREAD

**Discussion by shiyang1026** [LinuxSuRen/api-testing#359] [General]
@LinuxSuRen  ，您好，我遇到了一些关于开发环境问题

问题描述：我现在通过项目根目录的Dockerfile构建好了镜像，并且使用这个镜像构建并运行了容器，我可以正常 "使用" atest，但是我不知道应该如何使用docker的环境进行 "调试开发"，因为当我进入容器的命令行，我没有找到 `/workspace` 目录，当我运行 `go version`、`node -v` ，都显示 `command not found`，这让我有点疑惑 。

抱歉，我尝试了很多办法没能解决。

**Comment by LinuxSuRen** [comment]
首先，我推荐尽量在本地环境（指的是你的当前操作系统环境，例如：Windows、MacOS或 Linux），理由是：这种方式最简单，容易上手。当然，肯定也有远程调试、测试的方法，但鉴于当前项目复杂度，其实不太需要。

其次，对于开发环境来说，主流操作系统 Go 都是支持的。个人体验是，MacOS 是最流畅的，WSL 或者 Linux 也不错；对于 Windows 来说有些不是很习惯。而搭建 Go 的开发环境并不是很复杂，只要是把 Go 的安装包下载并安装好，基本就可以开始了。比较烦人的是，如果网络环境不是很好，或者没有“梯子”的话，首次编译项目会有些慢（主要是下载依赖的过程）。

对于 IDE 而言，GoLand 无疑是最推荐用来开发 Go 项目的，对于前端部分而言 WebStorm 也是一个非常不错的选择。我个人会在 VS-Code 和 GoLand 之间混合交替使用，各有优劣。

 > 当我进入容器的命令行，我没有找到 /workspace 目录，当我运行 go version、node -v ，都显示 command not found

关于这一点，你了解 Docker 的[多阶段（multi-stage）构建](https://docs.docker.com/build/building/multi-stage/)后就明白了。这里我简单说明下，在 Dockerfile 中有多个 `FROM` 语句，分别是用来编译 Go 代码、Vue 代码的，这种算是构建环境（阶段），而最后一个则是一个比较小的镜像，里面只保留和运行有关的文件。Go 代码编译后是可执行二进制文件，和 Java 不同（需要依赖 JVM 虚拟机运行），Go 程序是直接编译为不同架构、操作系统下的可执行程序。因此，你进入构建好的容器里找不到 Go 或者 NodeJS 环境是正常的。

**Reply by shiyang1026** [reply]
原来是这样，感谢解答！

**Reply by shiyang1026** [reply]
@LinuxSuRen 你好，仍然是一个关于环境的问题。

- 我阅读前端代码的时候发现有许多 import 的报错，可是我 `npm install` 之后明明所有的依赖都安装成功了，是我有哪里遗漏了吗？我的开发环境是 `Ubuntu 22.04.4 LTS` ，VSCode SSH Remote 到虚拟机的。

- 另外，我注意到这个关于暗色主题的 PR #247  很早之前已经被 Merge 了，但是我在本地运行没有这个功能(我 clone 的是最新的 master 分支)，这好像也不太正常


![image](https://github.com/LinuxSuRen/api-testing/assets/99083830/c7b7ce3a-888c-4965-bb3f-9bcfe9909586)

**Reply by LinuxSuRen** [reply]
你执行 `npm run build-only` 看看会不会报错，如果没错的话就只是语法提示的问题，可以忽略。

暗黑模式之前应该是可以的，可能是在之后修改的时候改坏了，你可以研究下按钮为啥消失了。
